### PR TITLE
fixed issue #4878 by updating assemble.py file

### LINKED ISSUE #4878: Add informative error message when the argument "shots" passed to the execute function is not of type int
### What is the expected enhancement?
execute(circuit, backend = Aer.get_backend('qasm_simulator), shots = "1024") doesn't give an informative error message (Issue number 842 in qiskit-aer). Add an informative error message.

This isssue was posted in qiskit-aer which is a wrong place for it. Creating a new issue at qiskit-terra (which is relevant to it) and creating a pull request for a solution

### PR DESCRIPTION
**Pull request by Arijit02**
### Summary

Fixes #4878 

### Details and comments

I have updated the assemble.py file under qiskit/compiler/ by adding a conditional statement to check whether argument shots is of type int or not. I raise a TypeError whenever shots is not int. I have also performed make style, make lint and make test in order to verify that I have not broken anything inside the project.

### CODE REVIEW

File: qiskit/compiler/assemble.py [hunk 1]
```diff
@@ -219,6 +220,9 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
             shots = min(1024, max_shots)
         else:
             shots = 1024
+    elif not isinstance(shots, int):
+        raise TypeError(
+            "Argument 'shots' should be of type 'int'")
```
Comment (taalexander) [inline qiskit/compiler/assemble.py:225]:
```suggestion
    elif not isinstance(shots, int):
        raise QiskitError(
            "Argument 'shots' should be of type 'int'"
     )
```

**Review by taalexander — CHANGES_REQUESTED** [review]
Could you please add a test in `test_assemble` to verify this raises as expected?

**Comment by Arijit02** [comment]
Yes, I can surely try to add a test in test_assembly

File: qiskit/compiler/assemble.py (hunk 1 above)
Comment (Arijit02) [reply qiskit/compiler/assemble.py:225]: Could you just clarify whether I should raise TypeError or QiskitError?

**Comment by Arijit02** [comment]
What I am doing is that I am raising a TypeError whenever shots is not of type int and writing a separate test for this in test_assembler.py. Or will I raise a QiskitError instead of TypeError?

**Comment by Arijit02** [comment]
I have updated this pull request but 2 checks failed as you can see. Could you tell me why these checks failed?

File: qiskit/compiler/assemble.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/compiler/assemble.py:226]: @taalexander suggests `QiskitError`.

Comment (Arijit02) [reply qiskit/compiler/assemble.py:226]: Ohk thanks, I got it.

**Review by taalexander — APPROVED** [review]

**Pull request merged**